src/test/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeFormatParserTest.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.ccd.domain.service.processor;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+class DateTimeFormatParserTest {
+
+    private DateTimeFormatParser dateTimeFormatParser;
+
+    @BeforeEach
+    void setUp() {
+        dateTimeFormatParser = new DateTimeFormatParser();
+    }
+
+    @Test
+    void shouldConvertDateTimeToIso8601() {
+        final String dateTimeFormat = "HHmmssSSS dd/MM/yyyy";
+        final String value = "123059000 20/10/2000";
+
+        final String result = dateTimeFormatParser.convertDateTimeToIso8601(dateTimeFormat, value);
+
+        assertAll(
+            () -> assertThat(result, is("2000-10-20T12:30:59.000"))
+        );
+    }
```

Comment:
Please add test cases for convertDateToIso8601 and some failure cases as well for the below exceptions 
IllegalArgumentException and DateTimeParseException